### PR TITLE
Add Search Engine Lookup plugin

### PR DESCRIPTION
**Pull request by rdswift**
This PR is to add the Search Engine Lookup plugin to the list of Picard Plugins available.

### CODE REVIEW

File: plugins/search_engine_lookup/__init__.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+            if isinstance(cluster, Cluster):
+                parts = []
+                if 'albumartist' in cluster.metadata and cluster.metadata['albumartist']:
+                    parts.extend(str(cluster.metadata['albumartist']).split())
+                if 'album' in cluster.metadata and cluster.metadata['albumartist']:
+                    parts.extend(str(cluster.metadata['album']).split())
+                if parts:
+                    if config.setting["search_engine_lookup_extra_words"]:
+                        parts.extend(str(config.setting["search_engine_lookup_extra_words"]).split())
+                    url = ENGINES[config.setting["search_engine_lookup_provider"]] + quote_plus(' '.join(parts))
```
Comment (zas) [inline plugins/search_engine_lookup/__init__.py:59]: I'd move `config.setting[*]' outside of the loop, storing values in local variables.

File: plugins/search_engine_lookup/__init__.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        # Settings for search engine provider
+        temp = config.setting["search_engine_lookup_provider"]
+        if temp not in ENGINES:
+            temp = 'Google'
+        if temp == 'Bing':
+            self.ui.rb_bing.setChecked(True)
+        elif temp == 'DuckDuckGo':
+            self.ui.rb_duckduckgo.setChecked(True)
+        else:
+            self.ui.rb_google.setChecked(True)
```
Comment (zas) [inline plugins/search_engine_lookup/__init__.py:110]: Generating radio buttions would permit to make code more generic and remove hardcoded references to search engines. `ENGINES` could be used (and extended).

**Review by zas — APPROVED** [review]
LGTM.

**Review by phw — APPROVED** [review]

File: plugins/search_engine_lookup/__init__.py (hunk 2 above)
Comment (rdswift) [reply plugins/search_engine_lookup/__init__.py:110]: These are both excellent suggestions.  I'll chip away at them when I have time over the next couple of weeks and submit a pull request with the changes.  Do you see storing the engines in a persistent dictionary (which can be edited from the ui), and then building the radio buttons from there?

**Pull request merged**